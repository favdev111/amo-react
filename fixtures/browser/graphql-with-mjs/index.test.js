const {
  bootstrap,
  startDevelopmentServer,
  startProductionServer,
} = require('../../utils');
const puppeteer = require('puppeteer');

beforeEach(async () => {
  await bootstrap({ directory: global.testDirectory, template: __dirname });
  global.appDevPort = await startDevelopmentServer({
    directory: global.testDirectory,
  });
  global.appProdPort = await startProductionServer({
    directory: global.testDirectory,
  });
  // Wait for serve to boot up
  await new Promise(resolve => setTimeout(resolve, 1000));
});

// https://github.com/facebook/create-react-app/issues/5234
// https://github.com/facebook/create-react-app/pull/5258
describe('graphql with mjs entrypoint', () => {
  it('correctly bundles files in development', async () => {
    const browser = await puppeteer.launch({ headless: true });
    try {
      const page = await browser.newPage();
      await page.goto(`http://localhost:${global.appDevPort}/`);
      await page.waitForSelector('.Pokemon-Name-Data');
      const output = await page.evaluate(() => {
        return Array.from(
          document.getElementsByClassName('Pokemon-Name-Data')
        ).pop().innerHTML;
      });
      expect(output).toMatchSnapshot();
    } finally {
      browser.close();
    }
  });

  it('correctly bundles files in production', async () => {
    const browser = await puppeteer.launch({ headless: true });
    try {
      const page = await browser.newPage();
      await page.goto(`http://localhost:${global.appProdPort}/`);
      await page.waitForSelector('.Pokemon-Name-Data');
      const output = await page.evaluate(() => {
        return Array.from(
          document.getElementsByClassName('Pokemon-Name-Data')
        ).pop().innerHTML;
      });
      expect(output).toMatchSnapshot();
    } finally {
      browser.close();
    }
  });
});
