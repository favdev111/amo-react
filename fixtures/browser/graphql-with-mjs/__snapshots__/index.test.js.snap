// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql with mjs entrypoint correctly bundles files in development 1`] = `"Pikachu"`;

exports[`graphql with mjs entrypoint correctly bundles files in production 1`] = `"Pikachu"`;
